src/publishMain.ml

```diff
@@ -280,6 +280,23 @@ let get_metas force tmpdir dirs opams urls repos tag names version =
              missing_names);
       metas
   in
+  let metas =
+    List.fold_left (fun metas name ->
+        let name = OpamPackage.Name.of_string name in
+        match
+          List.partition (function
+              | {name = Some name'; _} ->
+                OpamPackage.Name.equal name name'
+              | {name = None; _} -> false)
+            metas
+        with
+        | [], _ ->
+          OpamConsole.error_and_exit `Not_found
+            "Excluded package '%s' not found"
+            (OpamPackage.Name.to_string name)
```

Comment:
Why errorring instead of a warning and continue ?

Reply:
this was a request from David (see above). I think it's fair to error on this since it avoid issues for users later down the line if they do not see the warning or if the opam-publish is used as part of an automated system

Reply:
lgtm! Indeed, i looked at the code, didn't see David comment.